### ISSUE TITLE
[clang][ARM]report error for big struct

### ISSUE THREAD

**Issue by CoTinker** [llvm/llvm-project#72503]
demo.c
```
struct A { unsigned a : 1; unsigned b : 1; };
struct B
{
  unsigned char c[0x40000000];
  unsigned char d[0x40000ff0];
  struct A e;
};
```
For this demo, gcc will report error such as:
```
<source>:7:1: error: type 'struct B' is too large
    7 | };
      | ^
```
https://godbolt.org/z/adr4EPhzv

clang is not report error.

Someone say that the maximum size of a structure in gcc is `PTRDIFF_MAX == 2^31 - 1`, I wonder if clang has any rules on the size of the structure.
https://stackoverflow.com/questions/7724790/are-there-any-size-limitations-for-c-structures/49206497#49206497
